hearthstone/simulator/core/discover_object.py

```diff
@@ -0,0 +1,31 @@
+import itertools
+import typing
+from typing import Union, List, Callable
+
+from hearthstone.simulator.core.cards import MonsterCard
+from hearthstone.simulator.core.hero import Hero
+from hearthstone.simulator.core.secrets import Secret
+from hearthstone.simulator.core.spell import Spell
+
+if typing.TYPE_CHECKING:
+    from hearthstone.simulator.core.player import DiscoverIndex, Player
+
+Discoverable = Union[MonsterCard, Hero, Spell, Secret]
+
+
+class DiscoverObject:
+    def __init__(self, items: List[Discoverable], discover_function: Callable[[Discoverable], None], dissolve_leftovers: bool):
```

Comment:
I think the DiscoverObject should also have an enum saying what type of discover it is. This will make writing bots easier.